### PR TITLE
Bug: AddressMapping: Order of items returned from AddressMapping.entriesFromColumnsSpan depends on strategy

### PR DESCRIPTION
**Pull request by thilgen**
### Context
`AddressMapping` is an abstract class that internally can use either `SparseStrategy` or `DenseStrategy`

`AddressMapping.entriesFromColumnsSpan()` allows the caller to request a range of cells based on a specified `ColumnsSpan`

**Items returned from `AddressMapping.entriesFromColumnsSpan()` should be in the same order regardless of which strategy is in use**

**Unfortunately `SparseStrategy` returns the results by iterating first across the columns and then across the rows**
```
  public* entriesFromColumnsSpan(columnsSpan: ColumnsSpan): IterableIterator<[SimpleCellAddress, CellVertex]> {
    for (const col of columnsSpan.columns()) {
      const colMapping = this.mapping.get(col)
      if (colMapping !== undefined) {
        for (const [row, vertex] of colMapping.entries()) {
          yield [simpleCellAddress(columnsSpan.sheet, col, row), vertex]
        }
      }
    }
  }
```
**and  `DenseStrategy` returns the results by iterating first across the rows and then across the columns**
```
  public* entriesFromColumnsSpan(columnsSpan: ColumnsSpan): IterableIterator<[SimpleCellAddress, CellVertex]> {
    for (let y = 0; y < this.height; ++y) {
      for (let x = columnsSpan.columnStart; x <= columnsSpan.columnEnd; ++x) {
        const vertex = this.getCellVertex(x, y)
        if (vertex) {
          yield [simpleCellAddress(columnsSpan.sheet, x, y), vertex]
        }
      }
    }
```
**Which results in the following results in the attached test case**

**AlwaysSparse Return Values**
```
["0","0","1","45"] // row 0 col 1
["0","1","1","46"] // row 1 col 1
["0","2","1","47"] // row 2 col 1
["0","0","2","48"] // row 0 col 2
["0","1","2","49"] // row 1 col 2
["0","2","2","50"] // row 2 col 2
```
**AlwaysDense Return Values**
```
["0","0","1","45"] // row 0 col 1
["0","0","2","48"] // row 0 col 2
["0","1","1","46"] // row 1 col 1
["0","1","2","49"] // row 1 col 2
["0","2","1","47"] // row 2 col 1
["0","2","2","50"] // row 2 col 2
```

**This PR changes the processing order of `DenseStrategy.entriesFromColumnsSpan()` to be column first so that it returns results in the same order as `SparseStrategy.entriesFromColumnsSpan()`**

### How did you test your changes?
`npm:test`

### Types of changes
- [ ] Breaking change (a fix or a feature because of which an existing functionality doesn't work as expected anymore)
- [ ] New feature or improvement (a non-breaking change that adds functionality)
- [X] Bug fix (a non-breaking change that fixes an issue)
- [ ] Additional language file, or a change to an existing language file (translations)
- [ ] Change to the documentation

### Related issues:
1. n/a

### Checklist:
- [ ] My change is compliant with the [OpenDocument](https://docs.oasis-open.org/office/OpenDocument/v1.3/os/part4-formula/OpenDocument-v1.3-os-part4-formula.html) standard.
- [ ] My change is compatible with Microsoft Excel.
- [ ] My change is compatible with Google Sheets.
- [X] My code follows the code style of this project.
- [ ] I described my changes in the [CHANGELOG.md](https://github.com/handsontable/hyperformula/blob/master/CHANGELOG.md) file.
- [ ] My changes require a documentation update.
- [ ] My changes require a migration guide.

### CODE REVIEW

**Comment by adrianszymanski89** [comment]
Hi @thilgen 

Thank you for your time and involvement in helping to improve Hyperformula.

File: test/dense-sparse-cols.spec.ts
Comment (sequba) [inline test/dense-sparse-cols.spec.ts:1]: Please, move this test to `address-mapping.spec.ts` file.

I suggest phrasing it like this:
```js
describe('AddressMapping', () => {
  it('entriesFromColumnsSpan returns the same result regardless of the strategy', () => {
    //...
  })
})
```

**Review by sequba — CHANGES_REQUESTED** [review]
I absolutely agree with this change. Just please, move the test to the `AddressMapping` suite.

**Comment by sequba** [comment]
@thilgen Have you verified if this change has any impact on HyperFormula behavior from the end-user perspective?

File: test/dense-sparse-cols.spec.ts
Comment (thilgen) [reply test/dense-sparse-cols.spec.ts:1]: Will do

Comment (thilgen) [reply test/dense-sparse-cols.spec.ts:1]: Done.

**Comment by thilgen** [comment]
@sequba 

> @thilgen Have you verified if this change has any impact on HyperFormula behavior from the end-user perspective?

No example as such - this issue was uncovered as I was adding missing code coverage (**tests FTW**).

However, I could see a scenario where a customer was
* Using `HyperFormula` for a moderate to large use case and was looking to improve performance
* They read this article on [Performance](https://hyperformula.handsontable.com/guide/performance.html)
* And begin experimenting with `chooseAddressMappingPolicy` in their `ConfigurationOptions`

And suddenly they start to experience data anomalies - perhaps around serialization/de-serialization or custom copy-paste workflows - where the different ordering begins to re-shape their data in a way that could be difficult to unwind?

**Comment by sequba** [comment]
@thilgen Thank you very much for your involvement in the project. We need to focus on the scope of the upcoming release, and currently, we have no capacity to review your pull request. I can assure you that we will check it out as soon as we can.

**Pull request merged**